EvidenceApi/V1/UseCase/UpdateEvidenceRequestStateUseCase.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using EvidenceApi.V1.Gateways.Interfaces;
+using EvidenceApi.V1.Domain.Enums;
+using System.Linq;
+using EvidenceApi.V1.UseCase.Interfaces;
+using EvidenceApi.V1.Boundary.Response.Exceptions;
+using EvidenceApi.V1.Domain;
+using System.Collections.Generic;
+
+namespace EvidenceApi.V1.UseCase
+{
+    public class UpdateEvidenceRequestStateUseCase : IUpdateEvidenceRequestStateUseCase
+    {
+        private readonly IEvidenceGateway _evidenceGateway;
+
+        public UpdateEvidenceRequestStateUseCase(IEvidenceGateway evidenceGateway)
+        {
+            _evidenceGateway = evidenceGateway;
+        }
+
+        public EvidenceRequest Execute(Guid id)
+        {
+            var evidenceRequest = _evidenceGateway.FindEvidenceRequest(id);
+            if (evidenceRequest == null)
+            {
+                throw new NotFoundException($"Cannot find an evidence request with ID: {id}");
+            }
+
+            if (evidenceRequest.DocumentTypes.ToArray().All(dt =>
```

Comment:
One way of making this, and the other statement below checking if any evidence requests are in _Uploaded_ state, a bit more readable is to extract the logic into 2 private methods. It's good practice to name them something descriptive, for example `allEvidenceRequestsAreApproved`. See [here](https://refactoring.guru/decompose-conditional) for more info